src/tree-sitter-language-mode.js

```diff
@@ -456,6 +456,11 @@ class TreeSitterLanguageMode {
   syntaxTreeScopeDescriptorForPosition (point) {
     const nodes = []
     point = Point.fromObject(point)
+
+    if (point.column > 0 && point.column === this.buffer.lineLengthForRow(point.row)) {
+      point.column--
+    }
```

Comment:
Since we're mutating this point, we should pass `true` for [the `copy` parameter to `Point.fromObject`](https://github.com/atom/text-buffer/blob/d10f7595e600c775bd8bf505d11d5ba7a02a1290/src/point.coffee#L36-L45) above, to make sure we're not mutating a point that might get reused after the call to `scopeDescriptorForPosition`.